### PR TITLE
Add end-to-end tests that use nuget.exe

### PR DESCRIPTION
**Pull request by joelverhagen**
Addresses a big chunk of https://github.com/NuGet/NuGetGallery/issues/3788.

Tests:

1. nuget.exe install specific version
1. nuget.exe install latest version
1. nuget.exe restore `PackageReference`

### CODE REVIEW

File: src/NuGet.Services.EndToEnd/Support/CommandRunner.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    {
+        public static async Task<CommandRunnerResult> RunAsync(
+            string fileName,
+            string workingDirectory,
+            string arguments,
+            ITestOutputHelper logger,
+            IDictionary<string, string> environmentVariables = null,
+            int timeOutInMilliseconds = 60000)
+        {
+            await Task.Yield();
```
Comment (skofman1) [inline src/NuGet.Services.EndToEnd/Support/CommandRunner.cs]: >Task.Yield [](start = 18, length = 10)

why is this needed?

File: src/NuGet.Services.EndToEnd/Support/CommandRunner.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                process = new Process();
+
+                process.StartInfo = psi;
+                try
+                {
+                    process.Start();
+                }
+                catch (Exception ex)
+                {
+                    throw new InvalidOperationException($"Could not start process at with file name: {fileName}", ex);
```
Comment (xavierdecoster) [inline src/NuGet.Services.EndToEnd/Support/CommandRunner.cs]: nit: typo? `process at with`

File: src/NuGet.Services.EndToEnd/Support/PackageDirectory.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+using System.IO;
+using System.Linq;
+
+namespace NuGet.Services.EndToEnd.Support
+{
+    public class PackageDirectory
+    {
+        private readonly string _directory;
+
+        public PackageDirectory(string directory)
```
Comment (xavierdecoster) [inline src/NuGet.Services.EndToEnd/Support/PackageDirectory.cs]: Perhaps rename to `path`?
As such, [line 31](https://github.com/NuGet/NuGet.Services.EndToEnd/pull/10/files#diff-743a38bd7e4fa0d7296ba3955c27784aR31) will also return `directory._path` instead of `directory._directory` :)

File: src/NuGet.Services.EndToEnd/Support/PushedPackagesFixture.cs [hunk 4]
```diff
@@ -71,7 +84,7 @@ public PushedPackagesFixture(IGalleryClient galleryClient, TestSettings testSett
                 }
 
                 var packageTypes = GetPackageTypes(requestedPackageType);
-
+
```
Comment (xavierdecoster) [inline src/NuGet.Services.EndToEnd/Support/PushedPackagesFixture.cs]: nit: remove whitespace

File: src/NuGet.Services.EndToEnd/Support/SemVer2TheoryAttribute.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Xunit;
+
+namespace NuGet.Services.EndToEnd.Support
+{
+    public class SemVer2TheoryAttribute : TheoryAttribute
```
Comment (xavierdecoster) [inline src/NuGet.Services.EndToEnd/Support/SemVer2TheoryAttribute.cs]: 👍

File: src/NuGet.Services.EndToEnd/Support/CommandRunner.cs (hunk 1 above)
Comment (joelverhagen) [inline src/NuGet.Services.EndToEnd/Support/CommandRunner.cs]: To ensure that the whole process execution happens on a different thread than the caller. Otherwise, all the code up until the first `await` will happen on the calling thread.

File: src/NuGet.Services.EndToEnd/Support/PushedPackagesFixture.cs (hunk 4 above)
Comment (joelverhagen) [inline src/NuGet.Services.EndToEnd/Support/PushedPackagesFixture.cs]: Fixed.

File: src/NuGet.Services.EndToEnd/Support/CommandRunner.cs (hunk 2 above)
Comment (joelverhagen) [inline src/NuGet.Services.EndToEnd/Support/CommandRunner.cs]: Fixed.

File: src/NuGet.Services.EndToEnd/Support/PackageDirectory.cs (hunk 3 above)
Comment (joelverhagen) [inline src/NuGet.Services.EndToEnd/Support/PackageDirectory.cs]: Fixed.
